### ISSUE TITLE
Ensure that InfluxRPC responses respect block size

### ISSUE THREAD

**Issue by e-dard** [influxdata/influxdb_iox#3721]
I need to do some double checking internally but am writing this up so it doesn't slip my mind. If I'm right it also potentially impacts our query correctness validation efforts.

I think (though I need to check) that in an InfluxDB world we would return at most 1000 values and timestamps in an `{Integer, Unsigned, Float, Boolean, String}Frame`. This would be due to the maximum batch size of a TSM block being 1000 values.

If this is the case then we should teach IOx to do the same, mainly because I don't want us to violate any invariants in query clients that are not explicitly stated in a spec, e.g., that they expect frame sizes to be at most 1000 values.

I took a quick look in the `serieset` code in IOx and I'm pretty sure we're not limiting the maximum size of value frames.

https://github.com/influxdata/influxdb_iox/blob/4e3f66ed077d5336ec8e9df0ebd7825d0a227cf5/query/src/exec/seriesset/series.rs#L164-L174

**Comment by alamb** [comment]
I agree with your assessment. To the best of my knowledge there is nothing in IOx that will limit the number of points / values in a frame

**Comment by alamb** [comment]
@onelson reported he saw the following error while testing with larger datasets. I suspect it may be related to trying to stuff the entire output back in a single block

```
Error: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (9644576 vs. 4194304)
```

**Comment by alamb** [comment]
cc @crepererum  this seems related to https://github.com/influxdata/influxdb_iox/pull/6094

**Comment by crepererum** [comment]
Closed by #6094 (although we chunk based on size, not based on "number of values").
